src/somef/utils/constants.py

```diff
@@ -54,12 +54,12 @@
 REGEXP_READTHEDOCS_MD = (
     r"\(\s*(https?://[^\s\)]+\.readthedocs\.io[^\s\)]*)\s*\)"
 )
-# REGEXP_READTHEDOCS_HTML = (
-#     r"<a[^>]+href=['\"](https?://[^\s\"']+\.readthedocs\.io[^\s\"']*)['\"][^>]*>"
-#     r"(?:\s|<[^>]+>)*"
-#     r"<img[^>]+src=['\"]https?://readthedocs\.org/projects/[^\s/]+/badge/[^\s\"']*"
-# )
-REGEXP_READTHEDOCS_HTML = ( r"<a[^>]+href=['\"](https?://[^\s\"']+)['\"][^>]*>" r"(?:\s|<[^>]+>)*" r"<img[^>]+src=['\"]https?://readthedocs\.org/projects/[^\s/]+/badge/[^\s\"']*" )
+
+REGEXP_READTHEDOCS_HTML = (
+    r"<a\b[^>]+?href=['\"](https?://[^'\"\s]+?)['\"][^>]*?>"  
+    r"(?:(?!</a>)[\s\S])*?"                                  
+    r"<img\b[^>]+?src=['\"]https?://(?:readthedocs\.org/projects/|img\.shields\.io/pypi/)[^'\"\s]*"
```

Comment:
This regular expression is potentially dangerous, and it may be slow.
Try this one instead:
```
r"""
    <a\b
        [^>]*\bhref=['"](https?://[^'"\s]+)['"]
        [^>]*>
    (?:
        [^<]+             
        |
        <(?!/a\b)[^>]*>   
    )*
    <img\b
        [^>]*\bsrc=['"]
        https?://(?:readthedocs\.org/projects/|img\.shields\.io/pypi/)
        [^'"\s>]+
        ['"]
    """
```